### ISSUE TITLE
index outside the array bounds for `nth` filter

### ISSUE THREAD

**Issue by rootkea** [Keats/tera#650]
What should happen if the index falls outside the array bounds?

1. return empty string OR
2. error

At present, `nth(n=-1)` gives error while `nth(n=100)` gives empty string (for array having length < 100). This is inconsistent behavior.

**Comment by Keats** [comment]
It gives an error because only positive integers are allowed and we're trying to cast `n` as `usize`. Otherwise it's always an empty string.

**Comment by rootkea** [comment]
That I understand.

But may be we should return empty string `if n < 0` (before cast)?

This makes it consistent with the case of `n > array_size - 1`.

So that, nth filter will return empty string if n falls outside the array bounds (in either direction).

**Comment by Keats** [comment]
I'm wondering if this behaviour of returning the empty string should be removed entirely in Tera v2.

**Comment by rootkea** [comment]
Instead throw an error like "Out of bounds"?

+1 to that!

**Comment by Keats** [comment]
yes, although probably with some kind of `required`/`default` parameters

**Comment by Keats** [comment]
Closing as part of the Tera v2 release. If this is still relevant in v2, please reopen — see the 
  migration guide: https://github.com/Keats/tera/blob/master/MIGRATION.md
